flint.h

```diff
@@ -293,7 +293,7 @@ typedef __mpfr_struct flint_mpfr;
 
 #define FLINT_MAX(x, y) ((x) > (y) ? (x) : (y))
 #define FLINT_MIN(x, y) ((x) > (y) ? (y) : (x))
-#define FLINT_ABS(x) ((slong)(x) < 0 ? (-(x)) : (x))
+#define FLINT_ABS(x) ((slong)(x) < 0 ? -((ulong) (x)) : (ulong) (x))
```

Comment:
We can't change the type of FLINT_ABS. There might be code that does `-FLINT_ABS(x)`.

There should probably be a separate macro (like `UI_ABS_SI` macro in arf.h) for `slong -> ulong` absolute value.

Reply:
Thanks, will probably fix this tonight. Let me know if anything else needs to be fixed!

Reply:
First of all, there is no instance of `-FLINT_ABS(x)` in the whole repository. From a mathematical perspective, I think this definition makes more sense as `| \cdot | : \RR \to \RR_{> 0}`. From a computer's perspective, it would make more sense as `-WORD_MIN` is not defined.

However, as you imply, this would not be nice toward users of FLINT as this breaks the interface. What do you think about introducing an internal header where we can define `ABS(x)` to be equivalent to Arb's `UI_ABS_SI`?

Trevlig Valborg!

Reply:
Grep immediately found one such instance in Arb:

    acb_modular/hilbert_class_poly.c:        arb_set_si(acb_realref(z), -FLINT_ABS(qbf[3 * a + 1]));

Reply:
PS. du är inte med i Cortègen?

Reply:
Jag är nog mer av en åskådare av Cortègen än en deltagare, men det är definitivt en lokal högtid ;-)

Reply:
Jeg er ikke sikker på, om jeg forstår.

Reply:
Ich spreche kein dänisch